pkg/server/metrics/metrics.go

```diff
@@ -138,13 +159,20 @@ func newServerMetrics() *ServerMetrics {
 		backend:           backend,
 		pendingDials:      pendingDials,
 		fullRecvChannels:  fullRecvChannels,
+		dialFailures:      dialFailures,
 	}
 }
 
 // Reset resets the metrics.
 func (a *ServerMetrics) Reset() {
 	a.latencies.Reset()
 	a.frontendLatencies.Reset()
+	a.connections.Reset()
+	a.httpConnections.Set(0)
```

Comment:
This is a gauge, why are we setting this to 0?

Reply:
Eh. I set it to 0 since it doesn't have a `Reset()` method. But I think I agree that doesn't really make sense. Removed.